src/compiler/crystal/codegen/call.cr

```diff
@@ -359,7 +359,11 @@ class Crystal::CodeGenVisitor
           end
           node.args.each_with_index do |node_arg, i|
             a_def_arg = a_def.args[i]
-            result = and(result, match_type_id(node_arg.type, a_def_arg.type, arg_type_ids[i]))
+            if autocast_literal?(node_arg)
+              result = and(result, int1(1))
```

Comment:
I guess you preferred the explicitness of `result && true` and have as many `and` as args, rather than left result untouched `if autocast_literal?(node_arg)`. Right?